packages/components/src/public/QueryInfo.ts

```diff
@@ -258,10 +259,31 @@ export class QueryInfo {
         return extraDisplayColumn;
     }
 
-    getLookupViewColumns(omittedColumns?: string[]): QueryColumn[] {
-        const lcCols = omittedColumns ? omittedColumns.map(c => c.toLowerCase()) : [];
-        return this.columns.filter(col => col.shownInLookupView && lcCols.indexOf(col.fieldKey.toLowerCase()) === -1)
-            .valueArray;
+    getLookupViewColumns(displayColumnFieldKey?: string): QueryColumn[] {
+        let cols: QueryColumn[] = [];
+        if (this.views.has(IDENTIFYING_FIELDS_VIEW_NAME)) {
+            this.views.get(IDENTIFYING_FIELDS_VIEW_NAME).columns.forEach(col => {
+                const qCol = this.columns.get(col.fieldKey.toLowerCase());
+                if (qCol) {
+                    if (col.title) {
+                        cols.push(new QueryColumn({ ...qCol, caption: col.title }));
+                    } else {
+                        cols.push(qCol);
+                    }
+                }
+            });
+        } else {
+            const lcDisplayColumnFieldKey = displayColumnFieldKey?.toLowerCase();
+            if (displayColumnFieldKey) {
+                cols.push(this.columns.get(lcDisplayColumnFieldKey));
```

Comment:
This may resolve as `cols.push(undefined)`. I recommend:

```tsx
const displayColumn = this.getColumn(displayColumnFieldKey);
let lcDisplayColumnFieldKey: string;

if (displayColumn) {
    lcDisplayColumnFieldKey = displayColumn.fieldKey;
    cols.push(displayColumn);
}
```

Reply:
Done.